### DISCUSSION TITLE
Cannot cancel a workflow re-run that has not yet queued.

### DISCUSSION THREAD

**Discussion by divyam234** [community/community#135366] [Actions]
### Select Topic Area

Bug

### Body

{
  "message": "Cannot cancel a workflow re-run that has not yet queued.",
  "documentation_url": "https://docs.github.com/rest/actions/workflow-runs#force-cancel-a-workflow-run",
  "status": "409"
}

Not able to cancel workflow rerun when github actions are disabled

**Comment by srijon57** [comment]
Hello, my friend.
It sounds like you're encountering an issue where you're unable to cancel a workflow re-run that hasn't yet queued, especially when GitHub Actions are disabled. This error message "Cannot cancel a workflow re-run that has not yet queued." with status 409 indicates a conflict, often meaning that the workflow re-run cannot be canceled because it hasn't entered the queue.

Here's what you can do:

Check Workflow Status: Ensure that the workflow is not in a state where it cannot be canceled (e.g., still initializing or waiting to be queued). Workflows that haven’t yet started can sometimes be difficult to cancel.

Enable GitHub Actions Temporarily: If GitHub Actions are disabled, you might need to re-enable them temporarily to allow the workflow to enter a cancellable state. Once it's enabled, try canceling the workflow again.

Force Cancel: If the workflow has queued but you still cannot cancel it, you can use the (https://docs.github.com/rest/actions/workflow-runs#force-cancel-a-workflow-run) to force cancel the workflow run.

Example:
curl -X POST \
-H "Accept: application/vnd.github.v3+json" \
-H "Authorization: token YOUR_GITHUB_TOKEN" \
https://api.github.com/repos/OWNER/REPO/actions/runs/WORKFLOW_RUN_ID/cancel
Reach Out to GitHub Support: If the issue persists and you’re unable to resolve it, consider reaching out to (https://support.github.com/contact) for further assistance.

Let me know if these worked for you!

**Reply by gianantoniopini** [reply]
I'm having the same (or very similar) issue . My workflow run has status "Queued" and when I try to cancel or force-cancel it via the REST API I receive the same response that @divyam234 receives:

```
{
"message": "Cannot cancel a workflow re-run that has not yet queued.",
"documentation_url": "https://docs.github.com/rest/actions/workflow-runs#force-cancel-a-workflow-run",
"status": "409"
}
```

**Reply by DHeinMW** [reply]
where you able to solve this issue somehow?

**Reply by divyam234** [reply]
It is fixed in recent github updates.

**Reply by SampsonCrowley** [reply]
no it's not. it's still an issue 2 years later. I have about 20 jobs in my repo permanently in limbo because they can neither be queued nor can they be canceled.

**Reply by sanneboshuizen** [reply]
Same here!
